### ISSUE TITLE
questions regarding cannon rotation not being entirely accurate to other create mechanisms

### ISSUE THREAD

**Issue by averycoolbean** [Cannoneers-of-Create/CreateBigCannons#369]
tldr, i have been trying to control a cannon using sequenced gearshifts in an attempt to get more finegrained controls

brute-forced figuring out the ratio by which you have to rotate the shafts to actually rotate the cannon and was pretty close i think, 
at least close enough to notice that particularly rotations by smaller increments would regularly be just a few degrees further than was input (confirmed using a mechanical bearing and identically set sequenced gearshift) whereas others were perfectly accurate (e.g. 45° pitch up was always accurate and what i used initially for testing, whereas 40° would be off by as much as 3°)

upon a cursory glance it does appear that the yawdiff and pitchdiff are calculated on the clientside (my guess is to improve responsiveness when handcranking, but im also really clueless regarding java and could be misreading things), could this be impacting the accuracy of gearshifts by applying just a tad of server-client latency to it?

if that idea is as wrong as im expecting it to be, im honestly just really curious to figure out what *is* causing the inaccuracies since other create mechanisms dont appear to be impacted

**Comment by winner32** [comment]
Apologies for my English.
I realized that the mechanical bearing doesn't have this problem, so I checked the code for it.
The reason is that it uses a variable called "sequencedAngleLimit" to limit its rotation, which makes it very precise, whereas the canon mount doesn't have this feature.
By the way, yawdiff and pitchdiff are for client-side rendering, I guess.

**Comment by winsrewu** [comment]
i apologize for my english, in advance
I checked the code for the mechanical bearing, and found that it uses a variable called "sequencedAngleLimit" to limit the rotation.
So I made a demo mod, for mc fabric 1.18.2, bigcannons 0.5.2a to fix this question
https://github.com/winner32/bigcannonsfix/releases/tag/0.0.1.3
(im new here, i really dont know how to create a pull request)

By the way, the client diff is for client-side rendering and visualization, I guess.
